# __Visualizacion de datos__

## Importar librerias

In [1]:
import pandas as pd

## Cargar datos

In [2]:
df = pd.read_csv('./data_in/inmigrantes_canada.csv')
df.head()

Pais Continente           Region  1980  1981  1982  1983  1984  \
0       Afganistán       Asia      Sur de Asia    16    39    39    47    71   
1          Albania     Europa    Sur de Europa     1     0     0     0     0   
2          Argelia     África  Norte de África    80    67    71    69    63   
3  Samoa Americana    Oceanía        Polinesia     0     1     0     0     0   
4          Andorra     Europa    Sur de Europa     0     0     0     0     0   

   1985  1986  ...  2005  2006  2007  2008  2009  2010  2011  2012  2013  \
0   340   496  ...  3436  3009  2652  2111  1746  1758  2203  2635  2004   
1     0     1  ...  1223   856   702   560   716   561   539   620   603   
2    44    69  ...  3626  4807  3623  4005  5393  4752  4325  3774  4331   
3     0     0  ...     0     1     0     0     0     0     0     0     0   
4     0     2  ...     0     1     1     0     0     0     0     1     1   

   Total  
0  58639  
1  15699  
2  69439  
3      6  
4     15  

[5 rows x 38 columns]

In [3]:
# dimensiones de los datos
df.shape

(195, 38)

> Los datos presentan 195 registros y 38 variables

## Acerca de los datos

Fuimos contratados como analistas de datos para trabajar en una consultoría que presta servicios de migración para colombianos que desean viajar a Canadá. En este proyecto, estaremos analizando un conjunto de datos que contiene información histórica sobre la inmigración en Canadá proveniente de todos los países del mundo desde el año 1980 hasta el año 2013.

Tenemos la columna país, la columna continente, la columna región, y la columna que lleva los números de los años entre 1980 y 2013. Tenemos 33 años de registro y el total de inmigrantes en Canadá durante ese periodo.


## Filtracion de datos 

In [4]:
# mostrar el tipo de dato de cada vaiable
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 38 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Pais        195 non-null    object
 1   Continente  195 non-null    object
 2   Region      195 non-null    object
 3   1980        195 non-null    int64 
 4   1981        195 non-null    int64 
 5   1982        195 non-null    int64 
 6   1983        195 non-null    int64 
 7   1984        195 non-null    int64 
 8   1985        195 non-null    int64 
 9   1986        195 non-null    int64 
 10  1987        195 non-null    int64 
 11  1988        195 non-null    int64 
 12  1989        195 non-null    int64 
 13  1990        195 non-null    int64 
 14  1991        195 non-null    int64 
 15  1992        195 non-null    int64 
 16  1993        195 non-null    int64 
 17  1994        195 non-null    int64 
 18  1995        195 non-null    int64 
 19  1996        195 non-null    int64 
 20  1997      

> Las variables Pais, Continente y Region son de tipo object, el resto son enteras.

> Todas las variables presentan 195 datos lo que indican que no hay valores faltantes.

Vamos a configurar como indice, los valores de Pais, para de esta forma centrarnos en la migracion de colombianos a Canada. 

In [5]:
df.set_index('Pais', inplace=True)
df

Continente            Region  1980  1981  1982  1983  1984  \
Pais                                                                           
Afganistán              Asia       Sur de Asia    16    39    39    47    71   
Albania               Europa     Sur de Europa     1     0     0     0     0   
Argelia               África   Norte de África    80    67    71    69    63   
Samoa Americana      Oceanía         Polinesia     0     1     0     0     0   
Andorra               Europa     Sur de Europa     0     0     0     0     0   
...                      ...               ...   ...   ...   ...   ...   ...   
Vietnam                 Asia  Sudeste Asiático  1191  1829  2162  3404  7583   
Sahara Occidental     África   Norte de África     0     0     0     0     0   
Yemen                   Asia   Asia Occidental     1     2     1     6     0   
Zambia                África   África Oriental    11    17    11     7    16   
Zimbabue              África   África Oriental    72   114   102    44    32   

                   1985  1986  1987  ...  2005  2006  2007  2008  2009  2010  \
Pais                                 ...                                       
Afganistán          340   496   741  ...  3436  3009  2652  2111  1746  1758   
Albania               0     1     2  ...  1223   856   702   560   716   561   
Argelia              44    69   132  ...  3626  4807  3623  4005  5393  4752   
Samoa Americana       0     0     1  ...     0     1     0     0     0     0   
Andorra               0     2     0  ...     0     1     1     0     0     0   
...                 ...   ...   ...  ...   ...   ...   ...   ...   ...   ...   
Vietnam            5907  2741  1406  ...  1852  3153  2574  1784  2171  1942   
Sahara Occidental     0     0     0  ...     0     1     0     0     0     0   
Yemen                18     7    12  ...   161   140   122   133   128   211   
Zambia                9    15    23  ...    91    77    71    64    60   102   
Zimbabue             29    43    68  ...   615   454   663   611   508   494   

                   2011  2012  2013  Total  
Pais                                        
Afganistán         2203  2635  2004  58639  
Albania             539   620   603  15699  
Argelia            4325  3774  4331  69439  
Samoa Americana       0     0     0      6  
Andorra               0     1     1     15  
...                 ...   ...   ...    ...  
Vietnam            1723  1731  2112  97146  
Sahara Occidental     0     0     0      2  
Yemen               160   174   217   2985  
Zambia               69    46    59   1677  
Zimbabue            434   437   407   8598  

[195 rows x 37 columns]

Generamos un nuevo datafame que contengan los anos y la cantidad de migrantes de Colombia.

In [6]:
# lista de string de los anos
anos = list(map(str, range(1980,2014)))

In [7]:
# filtro que considera los anos y su respectiva cantidad de migrantes colombianos
colombia = df.loc['Colombia', anos]
colombia

1980     266
1981     326
1982     360
1983     244
1984     235
1985     214
1986     257
1987     376
1988     352
1989     439
1990     614
1991     652
1992     582
1993     464
1994     375
1995     371
1996     381
1997     578
1998     929
1999    1306
2000    2259
2001    2965
2002    3283
2003    4318
2004    4566
2005    6424
2006    6535
2007    5357
2008    5452
2009    4652
2010    5218
2011    4366
2012    3741
2013    3631
Name: Colombia, dtype: object

Hemos generado una serie con los datos que buscamos. Vamos a convertirlos en un dataframe para una mejor manipulacion.

In [8]:
# diccionario con los datos a convertir
col_dict = {
    'Año': colombia.index.tolist(),
    'Inmigantes': colombia.values.tolist()
}

In [9]:
# creamos el dataframe
datos_col = pd.DataFrame(col_dict)
datos_col.tail()

Año  Inmigantes
29  2009        4652
30  2010        5218
31  2011        4366
32  2012        3741
33  2013        3631